ui/raidboss/data/05-shb/eureka/delubrum_reginae_savage.js

```diff
@@ -596,6 +611,454 @@ export default {
+    {
+      id: 'DelubrumSav Avowed Brand Collect',
+      // These come from Environment, E0000000
+      // 8E5 Hot Brand: +1
+      // 8F3 Hot Brand: +2
+      // 8F4 Cold Brand: +1
+      // 8F8 Cold Brand: +2
+      netRegex: NetRegexes.gainsEffect({ effectId: ['8E5', '8F3', '8F4', '8F8'] }),
+      condition: Conditions.targetIsYou(),
+      run: (data, matches) => {
+        const brand = {
+          '8E5': 1,
+          '8F3': 2,
+          '8F4': -1,
+          '8F8': -2,
+        };
+        data.currentBrand = brand[matches.effectId];
+      },
+    },
+    {
+      id: 'DelubrumSav Avowed Hot And Cold Unwavering Apparations',
+      // The buffs come out before the spell cast
+      // Trinity Avowed and/or Avowed Avatar receive one of these buffs:
+      // 08F9: Hot Blade: +1
+      // 08FA: Hot Blade: +2
+      // 08FB: Cold Blade: -1
+      // 08FC: Cold Blade: -2
+      // Positional data in statusEffectsParams is often not up to date, use promise
+      //
+      // Trigger delayed until after Blade Of Entropy happens about ~100ms after
+      // to get left/right cleave info
+      netRegex: NetRegexes.gainsEffect({ effectId: ['08F9', '08FA', '08FB', '08FC'] }),
+      delaySeconds: 0.3,
+      suppressSeconds: 1,
+      promise: async (data, matches, output) => {
+        const trinityLocaleNames = {
+          en: 'Trinity Avowed',
+          de: 'Trinität Der Eingeschworenen',
+          fr: 'Trinité Féale',
+          ja: 'トリニティ・アヴァウ',
+        };
+
+        const avatarLocaleNames = {
+          en: 'Avowed Avatar',
+          de: 'Spaltteil der Eingeschworenen',
+          fr: 'Clone De La Trinité Féale',
+          ja: 'アヴァウドの分体',
+        };
+
+        // select the Trinity and Avatars
+        let combatantNameBoss = null;
+        let combatantNameAvatar = null;
+        combatantNameBoss = trinityLocaleNames[data.parserLang];
+        combatantNameAvatar = avatarLocaleNames[data.parserLang];
+
+        let combatantDataBoss = null;
+        let combatantDataAvatars = null;
+        if (combatantNameBoss) {
+          combatantDataBoss = await window.callOverlayHandler({
+            call: 'getCombatants',
+            names: [combatantNameBoss],
+          });
+        }
+        if (combatantNameAvatar) {
+          combatantDataAvatars = await window.callOverlayHandler({
+            call: 'getCombatants',
+            names: [combatantNameAvatar],
+          });
+        }
+
+        // if we could not retrieve combatant data, the
+        // trigger will not work, so just resume promise here
+        if (!(combatantDataBoss !== null &&
+          combatantDataBoss.combatants &&
+          combatantDataBoss.combatants.length) ||
+          !(combatantDataAvatars !== null &&
+          combatantDataAvatars.combatants &&
+          combatantDataAvatars.combatants.length)) {
+          data.safeZone = null;
+          return;
+        }
+
+        // we need to filter for the Trinity Avowed with the lowest ID
```

Comment:
I think arguably you could figure this out from positional data too, if you're calling getCombatants? Are the ids here a known way this works?

Reply:
It is possible to get their position from X and Y, but from the few logs I viewed it's always been the same four combatant order so I just assumed they would be in the right spot. It's a bit of extra code to handle if they could be in different locations.

Reply:
Sounds good! Mostly just curious.  :)

Reply:
I am looking at more logs other than my own as mine always produced the same positions, I think I may have to add in code for to verify the location as I see logs on fflogs where they are not always in the same position.

https://www.fflogs.com/reports/zZgdKMLvjDB6aFwT#fight=7&hostility=1&view=replay
Trinity Avowed East, but the Avoweds appear to be in different locations assuming that FFlogs numbers are based on appearance order.

Reply:
I am going to continue to assume the boss always goes East. It auto moves to that position there before starting the mechanic.